charts/external-dns/README.md

```diff
@@ -0,0 +1,66 @@
+# ExternalDNS
+
+[ExternalDNS](https://github.com/kubernetes-sigs/external-dns/) synchronizes exposed Kubernetes Services and Ingresses with DNS providers.
+
+## Installing the Chart
+
+Before you can install the chart you will need to add the `external-dns` repo to [Helm](https://helm.sh/).
+
+```shell
+helm repo add external-dns https://kubernetes-sigs.github.io/external-dns/
+```
+
+After you've installed the repo you can install the chart.
+
+```shell
+helm upgrade --install external-dns/external-dns
```

Comment:
Would be helpful to have an example for another provdider like Hetzner or DigitalOcean.

They expect the Token via ENV.

```yaml
 env:
        - name: HETZNER_TOKEN
          value: "YOUR_HETZNER_DNS_API_KEY"
```

So to have it `GitOps` conform you should set the env like this:

```yaml
env:
            - name: DO_TOKEN
              valueFrom:
                secretKeyRef:
                  name: <NAME>
                  key: <KEY_NAME>
```
And should define a secretref in the values.yaml

Reply:
The current chart isn't actually configured to work with AWS, it just uses AWS as a valid argument so that it can be deployed to test the Kubernetes resources, there are no credentials. To use any provider the correct credentials must be provided, but this should be covered in the provider specific documentation and isn't specific to the Helm chart. For example AWS could be configured by an env mounted secret as seen in your DO example above, it could also be configured as a volume mounted secret, however it should be configured as a service account annotation using IRSA.

Reply:
i did not spot an option to set a custom env variable and mount and ref to it. Maybe i missed the place in the template code. I just need a way to support providers with API Tokens without havin to patch the external-dns deplyoment.

Reply:
You've got the `env` value which is in _values.yaml_ and documented in the chart README.

Reply:
ok i must missed this 😅 .

But i would add an example in the Readme, on how to use it that way. I would add it but i cant commit in your branch.

Otherwhise, i wait it gets merged and make a new PR. 

It would help to increase the DX and accessibility

Reply:
It's pretty idiomatic to well authored Helm charts, I'm happy to add a comment in the README values section but I don't think an example is appropriate here. If we start adding example for using idiomatic Helm patterns, especially ones which just capture the Kubernetes objects and pass them through as is the case here, we'd need an example of all the areas like this such as volumes and volume mounts etc and would be basically teaching people how to use Kubernetes & Helm.

Reply:
Yepp, some comment in the Readme and a hyperlink to the external-dsn main readme with the list of providers would be fine too! There its written how the provider expects som auth stuff.  Fine with this 😄 

Thanks for Steve for the discussion!

Reply:
I've updated the README with links to the relevant K8s APIs that are supported in the _values.yaml_ and linked the `provider` value to the README so I'm going to mark this thread as resolved.